clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -456,7 +456,7 @@ private boolean renameObject(LakeFSFileStatus srcStatus, Path dst) throws IOExce
                         creationReq);
             } catch (ApiException e) {
                 if (e.getCode() != HttpStatus.SC_INTERNAL_SERVER_ERROR ||
-                        !e.getMessage().contains("invalid API endpoint")) {
+                        !e.getResponseBody().contains("invalid API endpoint")) {
```

Comment:
The `getReponseBody()` method can return nil can you verify before calling `contains`.

Reply:
Done.